src/partitioning/block.jl

```diff
@@ -55,16 +57,28 @@ function partition(object, method::BlockPartition)
     append!(subsets[i], j)
   end
 
+  if calculate_metadata == false
+    println("Metadata calculation has been disabled. To calculate Metadata, please enable it while calling BlockPartition with neighbors = true.") 
+  end
+  #Intitialize metadata to an empty Dict.
+  #If metadata calculation is enabled, we will populate the Dict.
+  metadata = Dict()
+
   # neighboring blocks metadata
-  bstart  = CartesianIndex(ntuple(i -> 1, Dim))
-  boffset = CartesianIndex(ntuple(i -> 1, Dim))
-  bfinish = CartesianIndex(Dims(nblocks))
-  for (i, bcoords) in enumerate(bstart:bfinish)
-    for b in (bcoords - boffset):(bcoords + boffset)
-      if all(Tuple(bstart) .≤ Tuple(b) .≤ Tuple(bfinish)) && b ≠ bcoords
-        push!(neighbors[i], linear[b])
+  if calculate_metadata == true
+    println("Metadata Calculation has been enabled.")
+    bstart  = CartesianIndex(ntuple(i -> 1, Dim))
+    boffset = CartesianIndex(ntuple(i -> 1, Dim))
+    bfinish = CartesianIndex(Dims(nblocks))
+    for (i, bcoords) in enumerate(bstart:bfinish)
+      for b in (bcoords - boffset):(bcoords + boffset)
+        if all(Tuple(bstart) .≤ Tuple(b) .≤ Tuple(bfinish)) && b ≠ bcoords
+          push!(neighbors[i], linear[b])
+        end
       end
     end
+    # Save Calculated Metadata
+    metadata = Dict(:neighbors => neighbors)
```

Comment:
Since you did initialize the `metadata` before the if statement you could simply update it instead of creating a new one from scratch: `metadata[:neighbors] = neighbors`  or something like that.

Reply:
Working on it!